examples/actor_critic.py

```diff
@@ -0,0 +1,413 @@
+        done = done or self._steps_taken == FLAGS.episode_len
+
+        return self._state, reward, done, info
+
+    def reset(self):
+        self._env.reset()
+        self._steps_taken = 0
+        self._state = np.zeros(
+            (FLAGS.episode_len - 1, len(self.action_space)), dtype=np.int32
+        )
+        return self._state
+
+    def close(self):
+        self._env.close()
+
+
+class Policy(nn.Module):
+    """A very simple actor critic policy model."""
+
+    def __init__(self):
+        super().__init__()
+        self.affine1 = nn.Linear(
+            (FLAGS.episode_len - 1) * len(FLAGS.flags), FLAGS.hidden_size
+        )
+        self.affine2 = nn.Linear(FLAGS.hidden_size, FLAGS.hidden_size)
+        self.affine3 = nn.Linear(FLAGS.hidden_size, FLAGS.hidden_size)
+        self.affine4 = nn.Linear(FLAGS.hidden_size, FLAGS.hidden_size)
+
+        # Actor's layer
+        self.action_head = nn.Linear(FLAGS.hidden_size, len(FLAGS.flags))
+
+        # Critic's layer
+        self.value_head = nn.Linear(FLAGS.hidden_size, 1)
+
+        # Action & reward buffer
+        self.saved_actions: List[SavedAction] = []
+        self.rewards: List[float] = []
+
+        # Keep exponential moving average of mean and standard
+        # deviation for use in normalization of the input.
+        self.moving_mean = MovingExponentialAverage(FLAGS.mean_smoothing)
+        self.moving_std = MovingExponentialAverage(FLAGS.std_smoothing)
+
+    def forward(self, x):
+        """Forward of both actor and critic"""
+        # Initial layer maps the sequence of one-hot vectors into a
+        # vector of the hidden size. Next layers stay with the same
+        # size and use residual connections.
+        x = F.relu(self.affine1(x))
+        x = x.add(F.relu(self.affine2(x)))
+        x = x.add(F.relu(self.affine3(x)))
+        x = x.add(F.relu(self.affine4(x)))
+
+        # actor: choses action to take from state s_t
+        # by returning probability of each action
+        action_prob = F.softmax(self.action_head(x), dim=-1)
+
+        # critic: evaluates being in the state s_t
+        state_values = self.value_head(x)
+
+        # return values for both actor and critic as a tuple of 2 values:
+        # 1. a list with the probability of each action over the action space
+        # 2. the value from state s_t
+        return action_prob, state_values
+
+
+def select_action(model, state, exploration_rate=0.0):
+    """Selects an action and registers it with the action buffer."""
+    state = torch.from_numpy(state.flatten()).float()
+    probs, state_value = model(state)
+
+    # Create a probability distribution where the probability of
+    # action i is probs[i].
+    m = Categorical(probs)
+
+    # Sample an action using the distribution, or pick an action
+    # uniformly at random if in an exploration mode.
+    if random.random() < exploration_rate:
+        action = torch.tensor(random.randrange(0, len(probs)))
```

Comment:
fyi, i think you can use `torch.randint` here if you want, though it's probably an overkill and a bit of overhead for drawing a single number